aya/src/obj/btf/info.rs

```diff
@@ -0,0 +1,189 @@
+use std::{collections::HashMap, convert::TryInto};
+
+use bytes::BufMut;
+use object::Endianness;
+
+use crate::{
+    generated::{bpf_func_info, bpf_line_info},
+    obj::relocation::INS_SIZE,
+    util::bytes_of,
+};
+
+/* The func_info subsection layout:
+ *   record size for struct bpf_func_info in the func_info subsection
+ *   struct btf_sec_func_info for section #1
+ *   a list of bpf_func_info records for section #1
+ *     where struct bpf_func_info mimics one in include/uapi/linux/bpf.h
+ *     but may not be identical
+ *   struct btf_sec_func_info for section #2
+ *   a list of bpf_func_info records for section #2
+ *   ......
+ */
+#[derive(Debug, Clone, Default)]
+pub struct FuncSecInfo {
+    pub sec_name_offset: u32,
+    pub num_info: u32,
+    pub func_info: Vec<bpf_func_info>,
```

Comment:
Surely this needs to be an inline array and not a Vec? The kernel won't be able to work with Vec

Reply:
I used a vec here because we don't know the size ahead of time.
This struct isn't used verbatim by the kernel - it only cares about the `Vec<bpf_func_info>` and there is a `func_info_bytes()` function that does the conversion via `bytes_of`... I guess (not certain) that you could probably use `func_info_bytes.as_ptr()` - which would be nice, less code to maintain :tada: but I'd have to test it out.

Reply:
Never mind! I got confused see https://github.com/aya-rs/aya/pull/127#issuecomment-1004416147